src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -750,5 +750,23 @@ private static int ToInt32(IntPtr ptr)
 
             return (int)(ptr.ToInt64() & 0xffffffff);
         }
+
+        public void ShowTaskbarIcon(bool value)
+        {
+            var style = (UnmanagedMethods.WindowStyles)UnmanagedMethods.GetWindowLong(_hwnd, -20);
+            
+            style &= ~(UnmanagedMethods.WindowStyles.WS_VISIBLE);   
+
+            style |= UnmanagedMethods.WindowStyles.WS_EX_TOOLWINDOW;   
+            if (value)
+                style |= UnmanagedMethods.WindowStyles.WS_EX_APPWINDOW;
+            else
+                style &= ~(UnmanagedMethods.WindowStyles.WS_EX_APPWINDOW);
+
+            //Toggle to make the styles stick
+            UnmanagedMethods.ShowWindow(_hwnd, ShowWindowCommand.Hide);
```

Comment:
You should check if window is currently visible. It will be presented to user from hidden state othervise.

Reply:
Good point, I'll cache the previous state and then set it back to that